src/components/conversationalsearchresult/conversationalsearchresult.js

```diff
@@ -0,0 +1,164 @@
+import './conversationalsearchresult.scss';
+import handlebars from 'handlebars';
+import { marked } from 'marked';
+import { validateContainer } from '../../util/dom';
+import { observeStoreByKey } from '../../store';
+import PRECOMPILED_CONVERSATIONAL_SEARCH_RESULT_TEMPLATE from './precompile-templates/conversationalsearchresult.handlebars';
+import { registerHelper } from '../../util/handlebars';
+
+export default class ConversationalSearchResult {
+  constructor(client, reduxStore, conf) {
+    this.client = client;
+    this.conf = conf;
+    this.reduxStore = reduxStore;
+    this.answerMaxHeight = conf.answerMaxHeight || 150;
+    this.isSearchResultLoading = false;
+    this.observeResultLoadingState();
+
+    registerHelper('markdown', function (text) {
+      if (typeof text !== 'string') {
+        return '';
+      }
+
+      return new handlebars.SafeString(marked.parse(text, { breaks: true, gfm: true }));
+    });
+
+    if (validateContainer(conf.containerId)) {
+      observeStoreByKey(this.reduxStore, 'search', () => this.render());
+    }
+  }
+
+  observeResultLoadingState() {
+    this.reduxStore.subscribe(() => {
+      const isConversationalSearchResultLoading =
+        this.reduxStore.getState().search.loadingConversationalSearchResult;
+
+      if (this.isSearchResultLoading !== isConversationalSearchResultLoading) {
+        this.isSearchResultLoading = isConversationalSearchResultLoading;
+        this.render();
+      }
+    });
+  }
+
+  setupShowMoreButton() {
+    const answerContainer = document.querySelector('.answer-container');
+    const fadeOutOverlay = answerContainer.querySelector('.fade-out-overlay');
+    const showMoreBtn = document.querySelector('.show-more-btn');
+    const buttonText = showMoreBtn.querySelector('.button-text');
+    const chevron = showMoreBtn.querySelector('.chevron');
+
+    if (answerContainer && showMoreBtn) {
```

Comment:
Should all other DOM nodes be checked here too?
fadeOutOverlay, buttonText, chevron

Reply:
yes